### PR TITLE
Play 2.5 upgrade

### PR DESCRIPTION
**Pull request by rm-hull**

### CODE REVIEW

File: app/config/frontendGlobal.scala [hunk 1]
```diff
@@ -47,20 +49,20 @@ object FrontendGlobal
...
 object ControllerConfiguration extends ControllerConfig {
-  lazy val controllerConfigs = Play.current.configuration.underlying.as[Config]("controllers")
+  lazy val controllerConfigs: Config = Play.current.configuration.underlying.as[Config]("controllers")
 }
 
-object LoggingFilter extends FrontendLoggingFilter {
-  override def controllerNeedsLogging(controllerName: String) = ControllerConfiguration.paramsForController(controllerName).needsLogging
+object LoggingFilter extends FrontendLoggingFilter with MicroserviceFilterSupport {
+  override def controllerNeedsLogging(controllerName: String): Boolean = ControllerConfiguration.paramsForController(controllerName).needsLogging
 }
```
Comment (antroy) [inline app/config/frontendGlobal.scala]: Any reason for explicit types on this and controllerConfigs? Or just seeing what types they are without undoing again?

File: app/config/frontendGlobal.scala [hunk 2]
```diff
@@ -47,20 +49,20 @@ object FrontendGlobal
...
 
   override lazy val maskedFormFields = Seq("password")
 
   override lazy val applicationPort = None
 
   override lazy val auditConnector = FrontendAuditConnector
 
-  override def controllerNeedsAuditing(controllerName: String) = ControllerConfiguration.paramsForController(controllerName).needsAuditing
+  override def controllerNeedsAuditing(controllerName: String): Boolean = ControllerConfiguration.paramsForController(controllerName).needsAuditing
 }
```
Comment (antroy) [inline app/config/frontendGlobal.scala]: As above.

File: app/views/include/fieldError.scala.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@(errors: Seq[FormError], fieldName: String, onlyLast: Boolean = true)
+@(errors: Seq[FormError], fieldName: String, onlyLast: Boolean = true)(implicit messages: Messages)
 @if(errors.exists(_.key == fieldName)) {
     <p class="error-notification hard--bottom" data-field-error-@fieldName>
     @defining(errors.filter(_.key == fieldName)) { errors =>
```
Comment (antroy) [inline app/views/include/fieldError.scala.html]: Why implicit messages on this one, and just an explicit param on the others?

File: test/acceptance/BaseSpec.scala [hunk 4]
```diff
@@ -33,20 +34,8 @@ trait BaseSpec extends FeatureSpec with BeforeAndAfterAll with BeforeAndAfterEac
...
-        "microservice.services.auth.port" -> stubPort,
-        "microservice.services.third-party-application.host" -> stubHost,
-        "microservice.services.third-party-application.port" -> stubPort,
-        "microservice.services.third-party-developer.host" -> stubHost,
-        "microservice.services.third-party-developer.port" -> stubPort,
-        "microservice.services.api-definition.host" -> stubHost,
-        "microservice.services.api-definition.port" -> stubPort
-      ))
+  implicit override lazy val app: Application =
+    GuiceApplicationBuilder().configure("run.mode" -> "Stub").in(Mode.Prod).build()
```
Comment (antroy) [inline test/acceptance/BaseSpec.scala]: I like this - much nicer.

File: app/config/frontendGlobal.scala (hunk 1 above)
Comment (rm-hull) [inline app/config/frontendGlobal.scala]: IntelliJ suggests any public defs should have a type declaration - just getting rid of the yellow warning bars it was showing

File: app/config/frontendGlobal.scala (hunk 2 above)
Comment (rm-hull) [inline app/config/frontendGlobal.scala]: same answer as above

**Pull request merged**